source/component/PasDoc_Items.pas

```diff
@@ -1705,6 +1731,88 @@ procedure TPasItem.StoreParamTag(
   ReplaceStr := '';
 end;
 
+function TPasItem.InheritedItem: TPasItem;
+begin
+  if Assigned(MyObject) then
+    Result := MyObject.FindItemInAncestors(Name)
+  else
+    Result := nil;
+end;
+
+function TPasItem.GetInheritedItemDescriptions: TStringPairVector;
+
+  function IsOverrideItem(Item: TPasItem): Boolean;
+  begin
+    Result := (
+     ((Item is TPasRoutine) and (SD_OVERRIDE in TPasRoutine(Self).Directives)) or
+     ((Self is TPasProperty) and (TPasProperty(Self).PropType = '')));
+  end;
+
+  procedure AddAncestorDescription(Vector: TStringPairVector; ThisItemInAncestor: TPasItem);
+  begin
+    Vector.Add(TStringPair.Create(
+      ThisItemInAncestor.UnitRelativeQualifiedName,
+      ThisItemInAncestor.AbstractDescription + ThisItemInAncestor.DetailedDescription,
+      ThisItemInAncestor));
+  end;
+
+var
+  I: Integer;
+  CurrentClassAncestor: TPasCio;
+  InterfaceAncestor: TPasCio;
+  ThisItemInAncestor: TPasItem;
+  OverrideChainEnded: Boolean;
+begin
+  Result := TStringPairVector.Create(False);
+  OverrideChainEnded := not IsOverrideItem(Self);
+
+  if (Assigned(MyObject) and
+     (TPasCio(MyObject).MyType in [CIO_CLASS, CIO_PACKEDCLASS, CIO_OBJECT, CIO_PACKEDOBJECT])) then
+  begin
+    CurrentClassAncestor := MyObject;
+    while Assigned(CurrentClassAncestor) do
+    begin
+      { We want the main ancestor's description above all else, unless the method is not in
+        the inheritance tree of the method anymore. We want to keep traversing past that point
+        because we want to pick up descriptions from any interfaces. }
+      if not OverrideChainEnded then
+      begin
+        ThisItemInAncestor := CurrentClassAncestor.FindItem(Self.Name) as TPasItem;
```

Comment:
Write just `Name`, not `Self.Name`. We generally avoid writing explicit `Self` when it is not necessary.